src/client/Client.cc

```diff
@@ -4913,6 +4928,11 @@ void Client::handle_cap_import(MetaSession *session, Inode *in, const MConstRef<
     put_snap_realm(realm);
   
   if (in->auth_cap && in->auth_cap->session == session) {
+    if (!(wanted & CEPH_CAP_ANY_FILE_WR) ||
+	in->requested_max_size > m->get_max_size()) {
+      in->requested_max_size = 0;
```

Comment:
Okay this one's a bit more of a push than the others; why do we need to reset the requested_max_size whenever a cap import happens? Will we resend it to the new auth MDS in kick_flushing_caps I guess?

Reply:
because kick_flushing_caps() is noop if there is no dirty caps